Makes the Torch use Overmap

BlueNexus opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Things that  move:

- The Torch
- The Aquila
- The Calypso
- The Guppy

TODO
- [x] Helmsman job/alt-title/make the solgov pilot do it/idk
- [x] Make the vessel mobile and functional
- [x] Fix some minor pipe issues, work on fuel efficiency
- [x] Fix exploration shuttles
- [x] Decide where the engine control console goes
- [x] Make all current away sites places you can fly to - So you have to fly to an asteroid field and mine phoron to refuel.
- [x] Fix sideways flight
- [x] Map interiors for each engine pod, with access routes.
- [x] Warning-sign it up.
- [x] Add randomised event groups to the overmap

BlueNexus committed:
Overmap on the Torch

BlueNexus commented:
![](https://i.imgur.com/zYxWpcX.png)
Proof that Overmap actually works

babydoll commented:
i dont think we need more superfluous jobs, just let one of the 8 people already on the bridge do it

BlueNexus commented:
Could make it an alt title, aye.

BlueNexus commented:
Update - She moves!
https://i.imgur.com/sBWgSZA.gifv

ItsAnJonth commented:
Can I be r00d and ask if #15063 is still a thing/anyone knows anything about it

Finalsong commented:
Make it an alt-title of Bridge Officer.

LordLoko commented:
Helmsman could be a BO alt-title. If my 5 minute research on google is right, any rank can be a helmsman, from Enlisted to comissioned officers. So making the BO's be helmsman would not far from reality.

comma commented:
re: #15063 Iiii done quite a bit of shuttle rewriting after that, worked fine afterwards.

comma commented on code/modules/overmap/ships/ship.dm:
```diff
@@ -127,4 +127,8 @@
\ No newline at end of file
+		. |= E.can_burn()
+
+/obj/effect/overmap/ship/torch
```
ree put that stuff in map defeines

Also to fix sideways flight set fore_dir to WEST

ItsAnJonth commented:
@comma hail our lord and saviour, multitude-sprite-chimpanzee

BlueNexus commented:
Whyyyyyy are there still conflicts

BlueNexus commented:
I'll fix that later.

BlueNexus commented:
Progress update - Away missions are in. 

![](https://i.imgur.com/AqNukF7.png)

Crushtoe commented:
Don't make it an alt-title. Bridge Officers are actually useful, we don't need two people spawning in both with the helmsmen alt-title that refuse to do any Bridge Officer work because "I pilot." Just make it an assumed duty.

Kasuobes commented:
Did you even read the commit? It's not an alt title.

I would personally prefer if it's merged with the pilot job, allowing a total of two slots and actually giving pilots something to do. Make it a pilot alt title and remove Marines.

BlueNexus commented:
@Kasuobes Can't risk having two people with the helmsman alt-title.

Kasuobes commented:
Why not? One person does the actual piloting, the other watches the map to make sure you don't crash into stuff.

BlueNexus commented:
That's... Not how the helm console works. You watch the map /while/ you move around. - You move around on the overmap Z, and you're directly looking at that while flying.

Anyway, I decided on a new job.

Kasuobes commented:
I'm fine with the addition of the job, just sad that the solgov pilot still won't actually be useful.

BlueNexus commented:
This is ready for review, while I wrangle with mapmerge.

BlueNexus commented:
Ready for review.

BlueNexus commented:
MAP
MERGE
PY

BlueNexus commented:
Alright, I've committed the uncleaned map changes, because there is literally nothing I can do to make mapmerge work. I tried rebasing and redoing my changes, I tried copypasting over between map files, I tried manual resolution.

BlueNexus commented:
I literally cannot make mapmerge work, no matter what I do. I've tried everything - Rebasing to dev and redoing my changes, copypasting between maps, using the java plugin, using the python tool, trying to fix things by hand, etc. 

@PsiOmegaDelta @Techhead0 @comma Requesting permission to commit the uncleaned versions of the map files, which don't have corruption issues like so https://i.imgur.com/ynCZNuJ.png

BlueNexus commented:
This is DNM until someone figures out how to make mapmerge work.

Sorry, everyone.

BlueNexus commented:
![](https://i.imgur.com/ynCZNuJ.png)

BlueNexus commented:
Wrong button

Vivalas commented:
@BlueNexus REEE use FastDMM.

BlueNexus commented:
Many thanks to @pobiega for figuring out how to mapmerge this! This is ready for review.
Remove the DNM label please, @NullSnapshot

chaoko99 commented:
Would it be possible for antag shuttles to use this, or..?

BlueNexus commented:
@chaoko99 Yes, but they don't use it currently. Making them work properly would be tricky.

comma commented on maps/torch/torch_shuttles.dm:
```diff
@@ -1,3 +1,4 @@
+/* //In case we ever decide against overmap
```
If we do that we revert whole thing, not uncomment shuttles. Axe, not comment out.

BlueNexus replied:
Resolved.

comma commented:
Not seeing any issues overmapstuff-side.

Techhead0 commented on code/modules/atmospherics/pipes.dm:
```diff
@@ -364,6 +364,12 @@
+	color = PIPE_COLOR_ORANGE
+	maximum_pressure = 420*ONE_ATMOSPHERE
+	fatigue_pressure = 350*ONE_ATMOSPHERE
+	alert_pressure = 350*ONE_ATMOSPHERE
```
A) These are gonna be irreplaceable if removed somehow.
B) All the other colored pipes are just that, a color. If these are gonna be special, then the type-path should reflect that.

PsiOmegaDelta replied:
Concur.

BlueNexus replied:
I cannot correct this in the course of this PR, as I will have the same problems with mapmerge as previously, when editing paths in the map.

I'll correct this afterwards.

pobiega replied:
Sure you can. .dmm files are text, so you can just do a text-based replace for /obj/old/path to /obj/new/path.

BlueNexus replied:
Resolved.

Techhead0 commented on code/modules/atmospherics/pipes.dm:
```diff
@@ -613,6 +625,10 @@
+/obj/machinery/atmospherics/pipe/manifold/visible/orange
+	name = "Fuel pipe manifold"
+	color = PIPE_COLOR_ORANGE
+
```
The various fuel manifolds don't have the pressure resistance that the other pipes do.

BlueNexus replied:
Manifolds can't burst.

Techhead0 commented on maps/torch/job/jobs.dm:
```diff
@@ -296,6 +296,35 @@
+	)
+	allowed_ranks = list(
+		/datum/mil_rank/fleet/o2,
+		/datum/mil_rank/fleet/o1,
```
Lore issue, but if it's gonna be its own job, I honestly think it could be a petty officer job.

BlueNexus replied:
Leaving it like this unless @Raptor1628 says otherwise. I think it's appropriate, given the role has bridge access.

PsiOmegaDelta replied:
@Raptor1628

Raptor1628 replied:
Having been corrected by a member of the Navy, I'd say PO2, PO1, and CPO, all enlisted ranks.

Techhead0 commented on maps/torch/job/outfits.dm:
```diff
@@ -219,6 +219,21 @@ Keeping them simple for now, just spawning with basic EC uniforms, and pretty mu
 
+/decl/hierarchy/outfit/job/torch/crew/command/helmsman //until they get a better uniform
+	name = OUTFIT_JOB_NAME("Helmsman")
+	uniform = /obj/item/clothing/under/utility/expeditionary/pilot
```
I disagree with giving them the pilot pin. (The only difference in the `/pilot` subtype.) They're not small-craft pilots.

BlueNexus replied:
Will deal with this in a bit.

Techhead0 reviewed (commented):
Haven't read through the map changes yet, but here's my thoughts on the other things.

BlueNexus commented on maps/torch/job/outfits.dm:
```diff
@@ -219,6 +219,21 @@ Keeping them simple for now, just spawning with basic EC uniforms, and pretty mu
 
+/decl/hierarchy/outfit/job/torch/crew/command/helmsman //until they get a better uniform
+	name = OUTFIT_JOB_NAME("Helmsman")
+	uniform = /obj/item/clothing/under/utility/expeditionary/pilot
```
They're not small craft pilots, but they're still piloting the vessel. I feel like it's equivalent enough.

BlueNexus commented:
Going to come back to this tomorrow.

mustafakalash commented:
How does the Torch flying to away sites work? Does it just bring the sites into range for the shuttles to fly down?

Techhead0 commented:
>Does it just bring the sites into range for the shuttles to fly down?

From what I've heard, yeah. IIRC, the different shuttles have different ranges, and I think the GUP has the shortest range and can only land if the Torch is on the same grid square as the site.

mustafakalash commented:
Pretty neat. Awesome work, monkeyman and cirra. And any one else who did things. Neat all around.

BlueNexus commented:
Progress update - I decided to include on-overmap events in this PR, rather than in a second PR. It's almost finished, so this should be ready again soon.

PsiOmegaDelta commented on code/__defines/colors.dm:
```diff
@@ -46,4 +46,5 @@
-#define	PIPE_COLOR_BLACK	"#444444"
\ No newline at end of file
+#define	PIPE_COLOR_BLACK	"#444444"
+#define PIPE_COLOR_ORANGE	"#b95a00"
```
Minor but the indentation is inconsistent here.

BlueNexus replied:
It looks fine in the editor. I'll adjust it.

Techhead0 replied:
They're using tabs, not spaces.

BlueNexus replied:
Resolved.

PsiOmegaDelta commented on maps/torch/torch-2.dmm:
```diff
@@ -1832,6 +1832,75 @@
+"Kz" = (/obj/machinery/atmospherics/pipe/manifold4w/visible/orange,/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KA" = (/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTH)"; icon_state = "warning"; dir = 1},/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KB" = (/obj/machinery/light/small{dir = 8},/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTH)"; icon_state = "warning"; dir = 1},/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KC" = (/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTHEAST)"; icon_state = "warning"; dir = 5},/turf/simulated/floor/reinforced,/area/thruster/d3port)
```
Did you use that horrible "Generate from Icon States/Direction" option or did you acquire the warning decals from elsewhere on the ship?
I don't see these among the removed lines at least.

If it's the former then cleanup and remove all tag and icon_state assignments. The latter in particular is a real headache when icon state names change.
Repeat for all instances were you used the "Generate from" operation.

BlueNexus replied:
I never used generate from (I didn't even know that was an option). I acquired the decals from elsewhere.

BlueNexus replied:
This'll be fixed in Psi's mass-edit, later on.

PsiOmegaDelta reviewed (changes requested):
Some changes required.

BlueNexus commented on code/modules/atmospherics/pipes.dm:
```diff
@@ -364,6 +364,12 @@
+	color = PIPE_COLOR_ORANGE
+	maximum_pressure = 420*ONE_ATMOSPHERE
+	fatigue_pressure = 350*ONE_ATMOSPHERE
+	alert_pressure = 350*ONE_ATMOSPHERE
```
This will be corrected shortly.

BlueNexus commented on code/__defines/colors.dm:
```diff
@@ -46,4 +46,5 @@
-#define	PIPE_COLOR_BLACK	"#444444"
\ No newline at end of file
+#define	PIPE_COLOR_BLACK	"#444444"
+#define PIPE_COLOR_ORANGE	"#b95a00"
```
Will be corrected when I get home, after I push my overmap event stuff to this branch.

BlueNexus replied:
http://i.imgur.com/2obczVF.png

PsiOmegaDelta commented:
Also, there is a lack of images of the map changes.

BlueNexus commented:
@PsiOmegaDelta Some images are available on the forums. I'll grab those, as well as some new ones for areas not included in those screenshots (or which have outdated images on the forums), and put them here.

Most of the ones on the forums are outdated, so it'll be a bit.

BlueNexus commented on maps/torch/torch-2.dmm:
```diff
@@ -1832,6 +1832,75 @@
+"Kz" = (/obj/machinery/atmospherics/pipe/manifold4w/visible/orange,/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KA" = (/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTH)"; icon_state = "warning"; dir = 1},/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KB" = (/obj/machinery/light/small{dir = 8},/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTH)"; icon_state = "warning"; dir = 1},/turf/simulated/floor/reinforced,/area/thruster/d3port)
+"KC" = (/obj/effect/floor_decal/industrial/warning{tag = "icon-warning (NORTHEAST)"; icon_state = "warning"; dir = 5},/turf/simulated/floor/reinforced,/area/thruster/d3port)
```
DroneBL decided to disconnect me from IRC. What did you say, in regards to including this in your mass-edit?

BlueNexus commented:
Images:
http://i.imgur.com/xPYpKVL.png
http://i.imgur.com/EI92GN5.png
http://i.imgur.com/MviL2lb.png
http://i.imgur.com/WCV94Ud.png
http://i.imgur.com/0ZMz7qw.png
http://i.imgur.com/yqGGwwy.png

@PsiOmegaDelta

BlueNexus commented:
I need to take a break from this, my head hurts badly. I've kept going, and now I'm happy with it.

I hope this is satisfactory in its current form. All comments have been resolved, that can currently be resolved.

PsiOmegaDelta commented on code/modules/overmap/event.dm:
```diff
@@ -0,0 +1,67 @@
+	if(!start_y)
+		start_y = rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)
+
+	forceMove(locate(start_x, start_y, using_map.overmap_z))
```
So the event is placed on the map on new, then you go and relocate moments later in initialize?
Why just not keep it in null space?

BlueNexus replied:
I wasn't aware I /could/ keep it in nullspace.

PsiOmegaDelta commented on code/modules/overmap/event.dm:
```diff
@@ -0,0 +1,67 @@
+	if(!start_x)
+		start_x = rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)
+	if(!start_y)
+		start_y = rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)
```
```start_x = start_x || rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)```
```start_y = start_y || rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)```

PsiOmegaDelta commented on code/modules/overmap/event.dm:
```diff
@@ -0,0 +1,67 @@
+	..()
+
+/obj/effect/overmap/event/initialize()
+	if(!using_map.use_overmap)
```
The event should just CRASH() in new if ```using_map.use_overmap``` isn't true.

BlueNexus replied:
Resolved on the line below.

PsiOmegaDelta replied:
```initialize()``` isn't ```New()```.

PsiOmegaDelta commented on code/modules/overmap/event.dm:
```diff
@@ -0,0 +1,67 @@
+	if(sx)
+		start_x = sx
+	if(sy)
+		start_y = sy
```
Why just assign sx/sy immediately?
But if you're paranoid:
```
start_x = sx || start_x
start_y = sy || start_y
```

PsiOmegaDelta commented on code/modules/overmap/event.dm:
```diff
@@ -0,0 +1,67 @@
+/obj/effect/overmap/event/proc/trigger()
+	if(event_manager)
+		var/datum/event_meta/EM = new(EVENT_LEVEL_MODERATE, src.name, linked_event, add_to_queue = 0, is_one_shot = 1)
+		active_event = new linked_event(EM)
```
What if multiple ships enter the same turf?
The prior ```active_event```, if one is ongoing, is then de-referenced and won't be stopped on ```leave()```.

PsiOmegaDelta replied:
Seems you've given the ship the responsibility to not trigger multiple events/the same event multiple times.
I don't feel that's the appropriate location for it, thus the above comment still applies.

PsiOmegaDelta commented on code/modules/overmap/sectors.dm:
```diff
@@ -83,4 +83,15 @@
+			var/start_y = rand(OVERMAP_EDGE, using_map.overmap_size - OVERMAP_EDGE)
+			var/obj/effect/overmap/event/Ev = new E(locate(1, 1, 1), start_x, start_y)
+			for(var/attempt in 2 to Ev.count)
+				new E(locate(1, 1, 1))
```
Seems quite possible to spawn multiple events in the same location.

BlueNexus replied:
Resolved in event.dm

PsiOmegaDelta replied:
Seems you might as well supply ```null```, instead of wasting cycles acquiring a turf you're not staying in anyway.
Applies to this line as well as 2 up.

PsiOmegaDelta commented on code/modules/overmap/ships/ship.dm:
```diff
@@ -109,6 +110,25 @@
 			Move(newloc)
 		update_icon()
+		var/turf/t = loc
+		if(t.contents)
```
There is no need to check ```t.contents```.

comma replied:
This whole thing seems more fit for event object's Crossed proc, so they handle addition/removal of events rather than ship

PsiOmegaDelta commented on code/modules/overmap/ships/ship.dm:
```diff
@@ -109,6 +110,25 @@
+				if(valid)
+					current_events.Add(E)
+					E.trigger()
+		if(current_events)
```
When would current events ever be null under reasonable circumstances?

BlueNexus replied:
All the time, that the Torch isn't on an event tile?

PsiOmegaDelta commented on code/modules/overmap/ships/ship.dm:
```diff
@@ -109,6 +110,25 @@
+			for(var/obj/effect/overmap/event/Ev in current_events)
+				if(!(istype(E, Ev.type)))
+					Ev.leave()
+					current_events.Remove(Ev)
```
As stated elsewhere, I don't believe the ship object should be responsible for checking what happens when a turf is entered/left.

BlueNexus replied:
I wasn't able to make any other method of doing this work.

comma replied:
Instead of constantly checking it in processing, can stick it in event object's procs.

Uncrossed(atom/movable/O) is called when things would leave it.
Crossed(atom/movable/O) is when things step on it.

BlueNexus replied:
The ship no longer handles starting events, but still handles ending them.

PsiOmegaDelta replied:
Why? Events themselves can keep track of triggering objects, and stop in Uncrossed.
The ```current_events``` list basically isn't needed, and but a corresponding list could be kept by the events.

BlueNexus replied:
What if you move from a tile with one event into another tile with the same event?
Gets spammy.

PsiOmegaDelta replied:
> What if you move from a tile with one event into another tile with the same event?
> Gets spammy.

Good point. Make a system that can handle multi-turf covering events.

BlueNexus replied:
It would be too complex. Everything is placed at random, it would have to find all the events in range of it which are the same type, and group up, which would be too CPU Intensive for something so unimportant.

BlueNexus commented on code/modules/overmap/ships/ship.dm:
```diff
@@ -109,6 +110,25 @@
+			for(var/obj/effect/overmap/event/Ev in current_events)
+				if(!(istype(E, Ev.type)))
+					Ev.leave()
+					current_events.Remove(Ev)
```
Uncrossed isn't valid here. What if they move into another event of the same type? Would get spammy.
Crossed is doable, though.

BlueNexus replied:
If it's that much of a problem, I can change it, but it's a QOL thing.

BlueNexus replied:
I really don't want to do that right now, I feel like that can wait until after this PR, or at least after the playtest.

BlueNexus commented:
Alright, this is ready.

seekin4u commented:
@BlueNexus ready4what?
Good damn work, man!

Crushtoe commented:
Woo! Here we go merge!

BlueNexus commented:
Bump.

NullSnapshot commented:
DNM added. Slated for play-test

BlueNexus commented:
According to Snapshot, playtesting will occur once there aren't any further requested changes.
( @PsiOmegaDelta )

Techhead0 commented:
I don't have any changes to ask for before we see a playtest.

BlueNexus commented:
Will squash these commits later, on my phone and just want things to be ready for playtesting.

PsiOmegaDelta commented:
This is like the radiation code all over again.

BlueNexus commented:
That teaches me for trying to fix things on my phone. It should be fixed now, I'll squash it later, as that's not critical to the playtest.

PsiOmegaDelta committed:
Adds a semi-decent multi-turf overmap event handler.

Handles:
* Stopping/starting events only when the situation changes.
* Ensuring separation of concern.

Does not currently handle:
* Admins stopping overmap event.
* The overmap ship being deleted.
* Multiple overmap ships.

BlueNexus commented:
Thanks for the squash!

BlueNexus commented:
Update: I have redone the engine nacelles. It is now possible to toggle between Direct Fuel Mode, where fuel is pumped directly into the engines, and Combustion Mode, where fuel is pumped into a combustion chamber and burned before being pumped into the engines. Direct injection is more fuel efficient, combustion mode is faster. 

![](https://i.imgur.com/0L4uCxu.png)

The buttons for controlling engine mode are located in the engine monitoring room, next to the engine control console. I may change this if it isn't used.

BlueNexus commented:
Ready for the Test.

BlueNexus commented:
Ready for the Test.

LordLoko commented:
Some quick feedback I had from a test round
1)Helmsman should be command support and not head of staff in the manifesto

2)I had an weird bug, basically I clicked some things on the console, they keep loading, when I tried to close the console which resulted that I couldn't interact with it anymore. Other consoles worked fine but noo matter what it did the console was broke, the admin had to delete and spawna  new one.

PsiOmegaDelta commented on maps/torch/torch_unit_testing.dm:
```diff
@@ -6,58 +6,62 @@
-		/area/teleporter/seconddeck = NO_SCRUBBER|NO_VENT,,
 		/area/tcommsat/relay = NO_SCRUBBER|NO_VENT,
+		/area/teleporter/seconddeck = NO_SCRUBBER|NO_VENT,,
+		/area/thruster/d1port = NO_SCRUBBER,
```
Are there others thruster subareas than these listed?
If not these 4 lines could be shortened as ```/area/thruster = NO_SCRUBBER```.

BlueNexus replied:
I wasn't aware I could do that.

Will do so later.

BlueNexus commented:
Sorry about that hiccup. Ready to resume the test.

Techhead0 commented on code/modules/atmospherics/he_pipes.dm:
```diff
@@ -13,6 +13,10 @@ obj/machinery/atmospherics/pipe/simple/heat_exchanging
 
+	maximum_pressure = 360*ONE_ATMOSPHERE
+	fatigue_pressure = 300*ONE_ATMOSPHERE
+	alert_pressure = 360*ONE_ATMOSPHERE
```
Instead of just bandaiding numbers higher whenever a problem shows up, you could do stuff like... actually using passive pressure gates to control pressure?

BlueNexus replied:
This problem is caused by extremely hot supermatter setups. I can't fix this by adding pressure gates without greatly decreasing engine power, so this will have to do.

PsiOmegaDelta commented:
Include the updated tgui.css and tgui.js files when editing or adding new templates.

**Edit:** Nevermind, it's a NanoUI template.

BlueNexus commented:
@PsiOmegaDelta I have no idea what those are.

BlueNexus commented:
Updated nacelles: 

![](https://i.imgur.com/hB2eOy6.png)

BlueNexus commented:
Ready for further playtesting.

BlueNexus commented:
Bump.

comma commented on code/game/machinery/status_light.dm:
```diff
@@ -0,0 +1,38 @@
+/obj/machinery/status_light/initialize()
+	..()
+	update_icon()
+	spawn(45)
```
Gah, no, bad. I don't know where you copypasted that from, but no.
This hack was used in New() because you could never know if this object is created first, or is radio controllr created first.
Initialize is called after everything is created so this spawn is unneeded.

BlueNexus replied:
Noted.

ghost commented:
Weeew, im so hyped. @BlueNexus, you make a great work!

BlueNexus commented:
Ready again.

Chiquiii commented:
Conflictsss. :(

BlueNexus commented:
Resolved the conflicts.

mustafakalash commented on maps/torch/job/jobs.dm:
```diff
@@ -296,6 +296,35 @@
+	outfit_type = /decl/hierarchy/outfit/job/torch/crew/service/solgov_pilot
+	allowed_branches = list(
+		/datum/mil_branch/expeditionary_corps,
+		/datum/mil_branch/fleet = /decl/hierarchy/outfit/job/torch/crew/service/solgov_pilot/fleet,
```
Did you intentionally sneak in an undiscussed change of removing the marines from pilot?

BlueNexus replied:
No, i talked with raptor about whether marines could act as Helmsman (which is what the pilot does) and he said no.

Please ask for context before being accusatory in the future.

mustafakalash replied:
Asking Raptor is not discussing. Especially since you've been one to interpret his response to your liking in the past.

If the Sol pilot slot is only to be used for manning the helm, why not rename it to helmsman and completely replace the role?

BlueNexus replied:
It's not only for manning the helm, it's for piloting solgov-owned shuttles as well. Manning the helm is part of the SolGov Pilot's duties, as decided by a forum poll.

I asked Raptor because he's the loremaster, so the right person to ask about whether the people manning the helm could be marines lore-wise. I don't feel it needs a discussion, since the loremaster already decided on it.

I can always ask him again, if you want.

BlueNexus replied:
![](https://i.imgur.com/B5bjyh0.png)

@mustafakalash

mustafakalash replied:
But marines can fly the smaller ships.

BlueNexus replied:
I tried separating the jobs, and people did not like that whatsoever. Plus, the poll was clearly in favour of it being a SolGov Pilot's duty.

PsiOmegaDelta replied:
I'll be honest, this constantly dismissive attitude is why I down-prioritize bothering with your PRs.

BlueNexus replied:
Thanks for letting me know. I'll try and avoid that attitude in the future.

xales commented:
Cleaned up and reordered history, removing the merge, and put map changes in a separate commit.

BlueNexus commented:
That should have fixed the engine issues people were having.

BlueNexus commented:
Bump.

PsiOmegaDelta commented on code/game/machinery/buttons.dm:
```diff
@@ -201,3 +203,32 @@
+	..()
+	spawn(45)
+		if(radio_controller)
+			radio_connection = radio_controller.add_object(src, frequency, RADIO_ATMOSIA)
```
```
/obj/machinery/button/toggle/valve/initialize()
	..()
	radio_connection = radio_controller.add_object(src, frequency, RADIO_ATMOSIA)
```

PsiOmegaDelta commented on code/modules/overmap/ships/computers/helm.dm:
```diff
@@ -174,3 +174,65 @@
+	if (href_list["viewing"])
+		viewing = !viewing
+
+	add_fingerprint(usr)
```
```..()``` in Topic() already applies fingerprints as appropriate.

PsiOmegaDelta commented on code/modules/overmap/ships/computers/helm.dm:
```diff
@@ -174,3 +174,65 @@
+		viewing = !viewing
+
+	add_fingerprint(usr)
+	updateUsrDialog()
```
Not needed for NanoUI interfaces, if you correctly return 1 when an UI update is required.

PsiOmegaDelta commented on code/game/area/areas.dm:
```diff
@@ -6,6 +6,7 @@
 /area
 	var/global/global_uid = 0
 	var/uid
+	var/shuttle_locked = null
```
Do have some consideration for whether an extra var on a type is needed, or if a repository would be a better fit.

BlueNexus replied:
I don't think a new repository is needed in this case.

NullSnapshot replied:
Why is shuttle_locked being put in the base area class? Does every turf need the shuttle_locked var?
Do maintenance tunnels or solar areas need them?

BlueNexus replied:
Resolved

PsiOmegaDelta committed:
Overmap event manager now has a stricter sector selection.

Sectors which already contain other overmap effects are now ignored.
Also attempts some minor optimization of the predicate procs, now requiring fewer list instantiations.

BlueNexus commented:
@NullSnapshot Please either run a playtest, or remove the playtest requirement. At least reply to me when I ask about it, or even just acknowledge that you've seen my requests, rather than ignoring me.

EDIT: Resolved.

Chiquiii commented:
Dont give up, we're all rooting for you.

Kasuobes commented:
>@NullSnapshot Please either run a playtest, or remove the playtest requirement. At least reply to me when I ask about it, or even just acknowledge that you've seen my requests, rather than ignoring me. I'm losing my mind here, as well as my will to work on this.
😩

Oh hey, it's like my med overhaul all over again.

NullSnapshot commented:
> Oh hey, it's like my med overhaul all over again.

It's almost like big and contentious PRs with major feature changes that touch so many files that other stuff can't be merged until they're done are delicate things and we often times discourage such large PRs. Your med overhaul had almost a full month of playtesting on server but it was still too contentious for devs to agree whether we should keep it or not. This isn't the place to be making snarky comments about this kind of thing anyway.

BlueNexus committed:
Final tweaks, fixes, and comment resolution

Credit for making fuel pipes dispensable goes to @Pobiega, rest is by me.

BlueNexus commented on code/game/area/areas.dm:
```diff
@@ -6,6 +6,7 @@
 /area
 	var/global/global_uid = 0
 	var/uid
+	var/shuttle_locked = null
```
Resolved.

BlueNexus commented:
Fixes #16843

NullSnapshot commented:
@Baystation12/byond-game-devs please review at your convenience.

PsiOmegaDelta commented on maps/torch/torch_areas.dm:
```diff
@@ -482,6 +482,10 @@
 	requires_power = 1
 	lighting_use_dynamic = 1
 	flags = AREA_RAD_SHIELDED
+	var/shuttle_locked = "Calypso"
```
What's the purpose of this var when subtypes could just re-overload the proc?

BlueNexus replied:
I wanted to avoid hard-coding a return value into the procs, like `return "Calypso"`
If that's preferable, I'll do that, though.

BlueNexus replied:
Is that preferable, yes/no?

BlueNexus replied:
Resolved.

comma reviewed (approved):
Only iffy thing for me is area locking shuttles, but I'll get around to that in ressurection of harpy's landmark shuttles.
Otherwise just merge it already ree

orelbon commented:
marge plz.

Techhead0 commented:
Enough pre-merge tweaking and fussing. I think it's likely ready for live at this point as well.
